src/constraint_solver/io.cc

```diff
@@ -180,6 +188,16 @@ class FirstPassVisitor : public ModelVisitor {
   }
 
  private:
+  //// TODO: TBD: not sure we are confusing terms with IntExpr, losing IntVar-iness, nevermind Boolean, etc...
```

Comment:
I do not see your point

IntVar is a subclass of IntExpr.
The CP Solver creates IntExpr to represent arithmetic expressions (mostly)

expr = var_x + var_y

And expression is a min/max objects and offers events when the min or the max changes.
Constraints can subscribe to these events, and in many cases, this is enough.

IntVar are a subclass of IntExpr, it offers the same interface on min/max + value, removeValue and a few more methods.
BooleanVar are a special case of IntExpr and it should be invisible to the user.

Some constraints use the extended API (IntVar) and require thus variable as input.
The question is how to use x+y in a constraint (or any IntExpr).

You can create a special constraint Delegate(IntVar. IntExpr) that performs the z == x + y cast.
Then you can use z in the above constraint.

My point is that there are special cases delegate that are way more efficient that creating the expression first.  This is performed by the Var() call, and is registered in the CastInfo structure.

In the loader, it just tells you that the variable was created from the expression.
I do not understand how it affects you. 

You export the model, store the search information somewhere by name, load the model, scan to find the variables and reconstruct the search monitors.

And so far, in the 2 test I ran (SimpleTest and SimpleTestAfterSearch), expect the fact that I do not restore constraint names, they are working fine.